### PR TITLE
Optional orientation when spawning entity using spherical coordinates

### PR DESCRIPTION
**Pull request by chapulina**
# 🎉 New feature

* Follow up to #1008 
* Goes with this documentation change: https://github.com/ignitionrobotics/ign-msgs/pull/212

## Summary

#1008 added the ability of spawning an entity at a given latitude / longitude. But often, we also want to give the entity a custom 3D orientation (not just heading). With this PR, users can pass a `pose.orientation` that will be used in combination with lat / lon.

## Test it

See the added test, or try calling the service with a custom orientation.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: src/systems/user_commands/UserCommands.cc [hunk 1]
```diff
@@ -1118,12 +1119,21 @@ bool CreateCommand::Execute()
...
           math::SphericalCoordinates::LOCAL2);
 
-      auto poseComp = this->iface->ecm->Component<components::Pose>(entity);
-      *poseComp = components::Pose({pos.X(), pos.Y(), pos.Z(), 0, 0,
-          IGN_DTOR(createMsg->spherical_coordinates().heading_deg())});
+      // Override pos and add to yaw
+      createPose.SetX(pos.X());
+      createPose.SetY(pos.Y());
+      createPose.SetZ(pos.Z());
+      createPose.Rot().Euler(
```
Comment (mjcarroll) [inline src/systems/user_commands/UserCommands.cc:1126]: I think there could potentially be an issue here if you are pitched completely up or down, the euler angles will reach a singularity and I don't think that yaw addition will work as expected.

Comment (chapulina) [reply src/systems/user_commands/UserCommands.cc:1126]: Ah good point, yay Euler angles. Honestly I'd like to remove the heading, but I put it there in the beginning, so I think it should keep working the same way if the rest of the rotation is zero. The main question is what is even expected in this situation...

Comment (chapulina) [reply src/systems/user_commands/UserCommands.cc]: I'm using quaternion multiplication now, let me know what you think: 0fad8af908c51b33867f2d7986cfcf3b2d255afa

Comment (chapulina) [reply src/systems/user_commands/UserCommands.cc]: * And documented on https://github.com/ignitionrobotics/ign-msgs/pull/212

**Review by arjo129 — APPROVED** [review]
Tests seem to pass and the logic looks good to me.

File: src/systems/user_commands/UserCommands.cc (hunk 1 above)
Comment (mjcarroll) [reply src/systems/user_commands/UserCommands.cc]: Yes, much safer, thank you!

**Review by mjcarroll — APPROVED** [review]
LGTM with merge from main and green CI

**Comment by chapulina** [comment]
There was a sneaky bug detected by the touch plugin test, which was fixed in 10de2d1

**Review by mjcarroll — APPROVED** [review]

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-01-citadel-edifice-fortress/1313/1

**Pull request merged**